### PR TITLE
Include role honour allow duplicates

### PR DESCRIPTION
**Pull request by kiorky**
##### SUMMARY
include_role may re run roles that have allow_duplicates set to false in many cases, specially with 'imported' roles.
this pr brings consistency.
 
See associated test

This is part of the #32565 PR split.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
include_role

##### ANSIBLE VERSION
```
2.4
devel
```
testhost                   : ok=x   changed=0    unreachable=0    failed=x

```

Post (associatet PR test pass)
```
testhost                   : ok=x   changed=0    unreachable=0    failed=0

```

- related PRs: #35167 #35166 #35165 #35164 #35163

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/51752/38/tests

File: test/integration/targets/include_role_honour_duplicates/aliases [hunk 1]
```diff
@@ -0,0 +1,3 @@
+posix/ci/group3
+include_role
+include_role_honour_allow_duplicates
```
Comment (mattclay) [inline test/integration/targets/include_role_honour_duplicates/aliases]: Remove the include* aliases as they are not needed.

**Comment by kiorky** [comment]
!ready_for_review

**Comment by kiorky** [comment]
So i just rebased and on my fork i maintain patches for 2.4, 2.5 & 2.6

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/71713/27/tests

**Comment by kiorky** [comment]
errors seems unrelated to this PR changes.

**Comment by mattclay** [comment]
@kiorky I'm unable to reproduce the failure on the devel branch. The same error occurs on each attempted re-run of CI, so it does seem related to changes in this PR.

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/88123/32/tests

**Pull request closed without merging**